specification/k3po.control/SPEC.md

```diff
@@ -0,0 +1,244 @@
+may respond with one or more events or it may not respond at all (no event sent) when the action is finished.
+
+Throughout the flow description bellow, **an error** represents an ERROR event sent by the K3po driver.
+
+- The test framework initiates the conversation with a **PREPARE** command. This command will make the driver load and prepare a set
+of scripts for execution. The scripts are not yet executed, just loaded from disk and parsed. If there are ACCEPT instructions in the scripts,
+a bind on corresponding ports is performed. If the command is successful, the driver sends back a PREPARED event. In case of failure, an
+ERROR event is sent with the cause of the failure.
+
+- If the PREPARED command is successful, a **START** command may be sent by the test framework. This command will start the execution of the
+ clients in the scripts that were sent in the PREPARE command. If the command is successful, the driver sends back a STARTED event. This 
+ command will also trigger the sending of a FINISHED event whenever the scripts finished execution on the driver. In case of failure, an
+ERROR event is sent with the cause of the failure.
+
+- If there is no PREPARE command sent before the START command or if the PREPARE command was not successful an error with a message "Script has
+ not been prepared or is still preparing" must occur. 
+
+- If the START command was already sent, an error with a message "Script has already been started" must occur.
+
+- Any time after the PREPARE command and before the DISPOSE command, the test framework can send an **AWAIT** command for a barrier existing 
+in the scripts. The command receives a response with a NOTIFIED event only when the corresponding barrier is notified. The NOTIFIED event may
+ never be returned in case the barrier is never triggered. In case the barrier does not exist, an error must be returned.
+
+- Any time after the PREPARE command *and before DISPOSE?*, the test framework can send a **NOTIFY** command for a barrier existing in the 
+scripts. The command notifies the barrier and a response with a NOTIFIED event is received. In case the barrier does not exist, an error must 
+be returned.
+
+- Any time after the PREPARE command *and before DISPOSE?*, the test framework can send an **ABORT** command. The driver must abort the current 
+execution. If the execution of the scripts is already finished, no event is sent back. If the execution is aborted by this command, a FINISHED 
+event with the script that was executed so far must be returned.
+
+- Any time after the PREPARE command, the test framework can send a **DISPOSE** command. The driver must abort the execution and release the 
+resources in use. When done, a DISPOSED event must be sent back.
+
+
+TODO - Decision time :) - will we support multiple tests on the same connection or will we force closing the connection and reopening a new one ? 
```

Comment:
I recommend supporting via SPEC and implementing in driver.  I don't think we need to implement this in JUNIT.  It's actually much easier to debug in wireshark when connection for control protocol is closely associated with the tests streams it is running